crates/turborepo-cache/src/cache_archive/restore.rs

```diff
@@ -33,8 +34,9 @@ impl CacheReader {
 
     pub fn open(path: &AbsoluteSystemPathBuf) -> Result<Self, CacheError> {
         let file = path.open()?;
+        let is_compressed = path.extension() == Some("zst");
```

Comment:
Probably worth discussing here whether we want to check for the zstd magic number as well / instead / as a fallback? It'll be slower in the general case but then we are not dependent on the file name causing issues

Reply:
I'll defer to @gsoltis on that, but we should probably keep the same Go behavior until we're done with the port